Define some of the global styles in the theme

laurelfulford opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

As of #1724, the theme has dequeued WP 5.9's global styles, because the `!important` they include were causing unexpected issues. 

If you happen to enter one of the hex values from the colours used in the Global Styles in the block colour picker, it looks like the standard inline hex value is replaced with one of the Global Styles CSS classes on the front end. Because these classes are not defined, the colour is not applied on the front end. 

Though the odds are very low that this would happen for most of the hex values, but we've already had reports of issues with folks trying to use `#000`. I have included the full set of colour, background colour and border colour classes, on the very unlikely chance some stumbles upon a "luminous-vivid-orange" exact match. 

Closes #1748 

### How to test the changes in this Pull Request:

1. Add multiple group blocks to your editor, and give them different background colours -- #000, and a couple of the other 'unlikely' defaults -- like `#9b51e0` (vivid purple) and `#7bdcb5` (light green cyan):

![image](https://user-images.githubusercontent.com/177561/160723940-5fd77cbe-751b-4a3d-9e9a-c00a3fa673b6.png)

2. Publish your post; view on the front-end and note your colours aren't applied:

![image](https://user-images.githubusercontent.com/177561/160724084-ed94b094-3a0a-4a9d-bf16-6120a0189487.png)

4. Apply the PR and run `npm run build`. 
5. Confirm the colours are now applied:

![image](https://user-images.githubusercontent.com/177561/160724003-98d6a345-8a21-4397-bd96-c13d54efc1ba.png)

6. Test out the colours (primarily black) on other blocks (headers, buttons, pullquotes...) just to make sure it displays as expected. 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

adekbadek reviewed (commented):
I took the liberty of refactoring the code to reduce duplication in 959e520aeb37ddf765f892250b2561b55a620a51, let me know if that's ok and if I haven't messed it up while refactoring!

laurelfulford commented:
Ahhhh, that's so much better @adekbadek - thank you! 

I did a bunch of testing, and found two issues, but neither would have been introduced with the refactor: 

* There was a typo in the color-patterns.php file, so the primary border colour was always blue - before and after, and fixed in [305bc20](https://github.com/Automattic/newspack-theme/pull/1749/commits/305bc20fdacac56e6deabef8ccc4b119629181ae): 

![image](https://user-images.githubusercontent.com/177561/161841626-c51d4fc8-084d-402a-b3e9-6c351d2465bd.png)

![image](https://user-images.githubusercontent.com/177561/161842150-53a7eaf6-20ed-4b1f-bed0-9986951bd29c.png)


* The selector for the pullquote backgrounds wasn't specific enough to override a style from WordPress Core setting the `.wp-block-pullquote:not( .is-style-solid-color )` to `background: none` - before and after and fixed in [b52b9d5](https://github.com/Automattic/newspack-theme/pull/1749/commits/b52b9d5c8354ddddcb002e35b8075f140ea9c545). In WP 5.9 the global styles colours all use `!important`, which would explain why this wouldn't be an issue normally (unfortunately the `!important`s were causing us other issues!):

![image](https://user-images.githubusercontent.com/177561/161842004-25f1a4bc-0a56-4d3a-9f4d-7d6b150f90b9.png)

![image](https://user-images.githubusercontent.com/177561/161842349-3e18458a-a8b4-4e3c-8dfe-ff4f7c1237d2.png)

Could you take it for one last spin just to make sure things look okay?

adekbadek reviewed (approved):
Looks good after latest changes!
